### ISSUE TITLE
[feature] A friendlier output when pre-built packages are missing (precisely listing the config diffs).

### ISSUE THREAD

**Issue by p-groarke** [conan-io/conan#14810]
### What is your suggestion?

This one might be difficult, but it's also quite important for us.

If at all possible (I know already the answer is no), back-porting this feature to conan 1 would really help. We've been refactoring like crazy but we aren't even close to switching to conan 2 :)

Conan would do much better having a beginner friendly (first time user understandable) message when it cannot find pre-built packages for a given config / profile. It's hard to tell, even with configs you control yourself, what is clashing.

An example ideal output :
```
Hey buddy, we couldn't find any pre-built packages for your current configuration.
You may pass --build missing to build the package on your system.

Asked for config :
[settings]
bla
bla
bla
*flaflee* # in bold, closest matching configs dont have this option
[options]
bla
de
bla

Closest matching configurations found :
[settings]
bla
*notquitebla* # in bold, showing the difference
bla
[options]
bla
de
bla

# more profiles that are close, see below
```

Figuring out what configs are closest might be tricky, but also not that much.
You could go the simple route of comparing available configs, listing configs that have the least differences (multiple if more than one have the least amount of differences).

Once in a while it'll give weird results, for example a different OS maybe. But when that happens it should immediately raise flags and help us figure out the problem.

I know this is a big one, but we really *really* need something like this to stop relying on `--build missing`.
Thanks!

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Thanks for the feedback @p-groarke 

We are indeed discussing and trying to improve this UX, this is preliminary work in https://github.com/conan-io/conan/pull/14694. Still very early stage, but we are on it.

**Comment by p-groarke** [comment]
@memsharded That's great to hear! Don't feel like you ever need to respond on the week-ends :)

**Comment by memsharded** [comment]
Implemented in https://github.com/conan-io/conan/pull/14694, will be released next 2.0.15
